bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/LedgerStorage.java

```diff
@@ -60,6 +61,8 @@ void initialize(ServerConfiguration conf,
     void setStateManager(StateManager stateManager);
     void setCheckpointSource(CheckpointSource checkpointSource);
     void setCheckpointer(Checkpointer checkpointer);
+    void setMasterKeyCache(ConcurrentLongHashMap<byte[]> masterKeyCache);
```

Comment:
instead of exposing the internal implementation detail (ConcurrentLongHashMap) I would expose here something like:

```
interface LedgerStorageNotificationListener {
   void ledgerRemovedFromStorage(long ledgerId);
}
```

and here
`void addStorageStorageNotificationListener(LedgerStorageNotificationListener listener)`


and we don't need getters

Reply:
finished